src/main/scala/org/broadinstitute/dsde/workbench/leonardo/model/LeonardoModel.scala

```diff
@@ -130,36 +137,86 @@ case class ClusterInitValues(googleProject: GoogleProject,
                              jupyterProxySiteConf: GoogleBucketUri,
                              jupyterServerName: String,
                              proxyServerName: String,
-                             jupyterInstallExtensionScript: String,
-                             jupyterExtensionUri: GoogleBucketUri,
+                             jupyterInstallExtensionScript: GoogleBucketUri,
+                             jupyterExtensionUri: Option[GoogleBucketUri],
                              userServiceAccountCredentialsUri: GoogleBucketUri)
 
 
 object FirewallRuleRequest {
   def apply(googleProject: GoogleProject, dataprocConfig: DataprocConfig, proxyConfig: ProxyConfig): FirewallRuleRequest =
-    FirewallRuleRequest(name = dataprocConfig.clusterFirewallRuleName,
+    FirewallRuleRequest(
+      name = FirewallRuleName(dataprocConfig.clusterFirewallRuleName),
       googleProject = googleProject,
       targetTags = List(dataprocConfig.clusterNetworkTag),
-      port = proxyConfig.jupyterPort.toString,
+      port = proxyConfig.jupyterPort,
       protocol = proxyConfig.jupyterProtocol
     )
 }
 
-case class FirewallRuleRequest(name: String,
+case class FirewallRuleRequest(name: FirewallRuleName,
                                googleProject: GoogleProject,
                                targetTags: List[String] = List.empty,
-                               port: String, // could be a number "80" or range "5000-6000"
+                               port: Int,
                                protocol: String
                               )
 
+case class StorageObjectResponse(name: BucketObject,
+                                 bucketName: GoogleBucket,
+                                 timeCreated: Instant)
+
+
+
+
+
+//----------------------------------------------------------------------------
+// TODO remove after Rob merges his approach
```

Comment:
Referring to #41 here, which I assume I will have to rebase this PR onto

Reply:
So you should really just skip review of this section

Reply:
Thanks for marking this so clearly - I think mine is ready to merge, I just wanted to run through 1 more manual `create cluster -> jupyter -> delete cluster` test. Realistically should be done Monday